l1-contracts/src/core/Rollup.sol

```diff
@@ -253,6 +304,162 @@ contract Rollup is Leonidas, IRollup {
 
     emit L2ProofVerified(header.globalVariables.blockNumber, _proverId);
   }
+  //  TODO(#7346): Commented out for now as stack too deep (unused until batch rollups integrated anyway).
```

Comment:
Added a first impl of `verifyRootProof` for future use, but couldn't compile the contract due to stack too deep error. We don't currently use it and don't have the full capabilities in ts/sol to use it anyway, so I commented it out. I can also remove it and add back in future so we have cleaner code!

Reply:
How long do you expect it to be like this? I don't really like having a big bunch of uncommented code as it often end up just being distracting when reading through the code or something that is broken when one finally tries to un-comment it because things have changes and it have been standing still 🤷

Reply:
I agree - not sure on when. We will use it when batch rollups are fully integrated (the main part of the work will be editing the sequencer/orchestrator code, which is under Phil's team) after provernet is up. The logic follows from the other verify function pretty clearly so can remove for now for cleanliness.